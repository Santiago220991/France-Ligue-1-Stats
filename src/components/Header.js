import React from 'react';
import './Header.css';

const Header = () => (
  <div className="league-header">
    <img className="league-logo" src="https://media.api-sports.io/football/leagues/61.png" alt="league-logo" />
    <p className="season">2021-2022</p>
  </div>
);

export default Header;
