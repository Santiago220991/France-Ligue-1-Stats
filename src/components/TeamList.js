import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';
import './styles/TeamList.css';

const TeamList = () => {
  const [Filter, setFilter] = useState('');
  const Store = useSelector((store) => store.teams);

  const onChange = (e) => {
    const newFilter = e.target.value;
    setFilter(newFilter);
  };

  return (
    <div className="team-list">
      <div className="team-list-header">
        <p>TEAMS</p>
        <input className="search-input" onChange={onChange} type="text" placeholder="Search.." />
      </div>
      <div className="teams">
        {Store.filter((item) => {
          if (Filter === '') {
            return item;
          } if (item.team.name.toLowerCase().includes(Filter.toLowerCase())) {
            return item;
          }
          return false;
        }).map((element) => (
          <div className="team-element" key={element.team.name} id={element.team.name}>
            <Link className="link" to={`/TeamStats/${element.team.name}`}>
              <BsArrowRightCircle className="right-arrow" />
              <img src={element.team.logo} alt="team-logo" />
              <p>{element.team.name}</p>
              <p>
                Rank:
                {' '}
                {element.rank}
              </p>
            </Link>
          </div>
        ))}
      </div>
    </div>
  );
};
export default TeamList;
