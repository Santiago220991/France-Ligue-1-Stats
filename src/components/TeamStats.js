import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router';
import TableStats from './TableStats';
import './styles/TeamStats.css';

const TeamStats = () => {
  const { teamname } = useParams();
  const Store = useSelector((store) => store.teams);
  const team = Store.filter((element) => element.team.name === teamname);
  const teaminfo = team[0];
  if (teaminfo !== undefined) {
    const GoalDifference = teaminfo.all.goals.for - teaminfo.all.goals.against;
    return (
      <div className="TeamStats">
        <div className="Header-Team-info">
          <img className="stats-logo" src={teaminfo.team.logo} alt="team-logo" />
          <div className="general-states">
            <p className="team-name-stats">{teaminfo.team.name}</p>
            <p>
              Rank:
              {' '}
              {teaminfo.rank}
            </p>
            <p>
              Points:
              {' '}
              {teaminfo.points}
            </p>
            <p>
              Games Won:
              {' '}
              {teaminfo.all.win}
            </p>
            <p>
              Games Lost:
              {' '}
              {teaminfo.all.lose}
            </p>
            <p>
              Goal Difference:
              {' '}
              {GoalDifference}
            </p>
          </div>
        </div>
        <TableStats home={teaminfo.home} away={teaminfo.away} />
      </div>
    );
  }
  return <div className="error" />;
};

export default TeamStats;
