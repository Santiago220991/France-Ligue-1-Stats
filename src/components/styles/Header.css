.league-header{
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-left: 4%;
    padding-right: 4%;
}

.league-logo{
    height: 200px;
    width: 150px;
}

.season{
    font-size: 1.2rem;
    color: white;
}