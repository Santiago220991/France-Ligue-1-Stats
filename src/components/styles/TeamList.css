.team-list-header {
  display: flex;
  justify-content: space-between;
  padding-left: 4%;
  padding-right: 4%;
  align-items: center;
}

.team-list-header p {
  font-size: 1.1rem;
  color: white;
}

.search-input {
  height: 25px;
  width: 50%;
  border-radius: 4px;
}

.teams {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.err-msg {
  color: rgb(175, 6, 6);
  text-align: center;
}

.team-element {
  display: flex;
  flex-direction: column;
  text-align: center;
  flex-basis: 49%;
  animation-name: entering-left;
  animation-duration: 1s;
  border: solid white 1px;
}

.team-element:nth-child(4n-1) {
  background-color: #224586;
  animation-name: entering-right;
  animation-duration: 1s;
}

.team-element:nth-child(4n-2) {
  background-color: #224586;
  animation-name: entering-right;
  animation-duration: 1s;
}

.link {
  color: white;
  text-decoration: none;
  cursor: pointer;
  display: flex;
  flex-direction: column;
}

.right-arrow {
  margin-top: 5px;
  margin-right: 5px;
  align-self: flex-end;
}

.team-element p {
  margin-top: 0;
  margin-bottom: 5px;
}

.team-element img {
  width: 150px;
  height: 150px;
  align-self: center;
}

@keyframes entering-right {
  from {
    transform: translate(500px);
  }

  to {
    transform: translate(0);
  }
}

@keyframes entering-left {
  from {
    transform: translate(-500px);
  }

  to {
    transform: translate(0);
  }
}
