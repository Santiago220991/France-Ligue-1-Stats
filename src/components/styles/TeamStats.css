.Header-Team-info {
  display: flex;
  justify-content: center;
  align-items: center;
  animation-name: showing;
  animation-duration: 2s;
}

.stats-logo {
  width: 200px;
  height: 200px;
  margin-right: 2%;
}

.general-states {
  color: white;
}

.general-states p {
  margin-bottom: 0;
  margin-top: 0;
}

.team-name-stats {
  font-size: 24px;
  padding-bottom: 10px;
}

@keyframes showing {
  from {
    transform: scale(0);
  }

  to {
    transform: scale(1);
  }
}
