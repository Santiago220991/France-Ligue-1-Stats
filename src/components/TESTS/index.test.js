import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import Store from '../../redux/configureStore';
import Header from '../Header';
import TeamList from './__mocks__/TeamList';
import TeamStats from './__mocks__/TeamStats';
import { render } from '@testing-library/react';

describe('Testing React Components Rendering', () => {
  test('Testing Header Component', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Testing TeamList Component', () => {
    const tree = renderer.create(<Provider store={Store}><TeamList /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Testing TeamStats Component', () => {
    const tree = renderer.create(<Provider store={Store}><TeamStats /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('There should be a Team in the list called Paris Saint Germain', () => {
    const { container } = render(<Provider store={Store}><TeamList /></Provider>);
    expect(container).toHaveTextContent('Paris Saint Germain');
  });
  test('There should be a Team in the list called Paris Saint Germain', () => {
    const { container } = render(<Provider store={Store}><TeamList /></Provider>);
    expect(container).toHaveTextContent('Paris Saint Germain');
  });
  test('There should be a Team in the list called Marseille', () => {
    const { container } = render(<Provider store={Store}><TeamList /></Provider>);
    expect(container).toHaveTextContent('Marseille');
  });
  test('There should be a Team in the list called Paris Saint Germain with rank 1', () => {
    const { container } = render(<Provider store={Store}><TeamList /></Provider>);
    expect(container).toHaveTextContent('Paris Saint Germain');
    const rank=container.querySelector(".team-element")
    expect(rank).toHaveTextContent("Rank: 1")
  });
});
