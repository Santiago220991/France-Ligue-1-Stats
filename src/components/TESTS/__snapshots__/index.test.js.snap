// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing React Components Rendering Testing Header Component 1`] = `
<div
  className="league-header"
>
  <img
    alt="league-logo"
    className="league-logo"
    src="https://media.api-sports.io/football/leagues/61.png"
  />
  <p
    className="season"
  >
    2021-2022
  </p>
</div>
`;

exports[`Testing React Components Rendering Testing TeamList Component 1`] = `
<div
  className="team-list"
>
  <div
    className="team-list-header"
  >
    <p>
      TEAMS
    </p>
    <input
      className="search-input"
      placeholder="Search.."
      type="text"
    />
  </div>
  <div
    className="teams"
  >
    <div
      className="team-element"
    >
      <svg
        className="right-arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        />
      </svg>
      <img
        alt="team-logo"
        src="https://media.api-sports.io/football/teams/85.png"
      />
      <p>
        Paris Saint Germain
      </p>
      <p>
        Rank:
         
        1
      </p>
    </div>
    <div
      className="team-element"
    >
      <svg
        className="right-arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        />
      </svg>
      <img
        alt="team-logo"
        src="https://media.api-sports.io/football/teams/81.png"
      />
      <p>
        Marseille
      </p>
      <p>
        Rank:
         
        2
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing React Components Rendering Testing TeamStats Component 1`] = `
<div
  className="TeamStats"
>
  <div
    className="Header-Team-info"
  >
    <img
      alt="team-logo"
      className="stats-logo"
      src="https://media.api-sports.io/football/teams/85.png"
    />
    <div
      className="general-states"
    >
      <p
        className="team-name-stats"
      >
        Paris Saint Germain
      </p>
      <p>
        Rank:
         
        1
      </p>
      <p>
        Points:
         
      </p>
      <p>
        Games Won:
         
        26
      </p>
      <p>
        Games Lost:
         
        4
      </p>
      <p>
        Goal Difference:
         
        54
      </p>
    </div>
  </div>
</div>
`;
